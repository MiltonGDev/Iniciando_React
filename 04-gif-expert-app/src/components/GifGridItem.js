import React from 'react'

export const GifGridItem = ({img, tittle, url}) => {
    return (
        <div className='card animate__animated animate__fadeIn'>

            <img src={url} alt={tittle}></img>
            <p>{tittle}</p>

        </div>
    )
}
