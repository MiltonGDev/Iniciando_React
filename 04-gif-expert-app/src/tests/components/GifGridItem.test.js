import React from "react";
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import  { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas para GifGridItem', () => {
    
    const url = 'https://media2.giphy.com/media/yo3TC0yeHd53G/giphy.gif?cid=446fbf6ca4d0d24cygq8kkdjokh6d3w8qdz4otn5u21ealob&rid=giphy.gif&ct=g';
    const tittle = 'A tittle'; 
    let wrapper = shallow( <GifGridItem tittle={tittle} url={url}/> );

    test('debe de mostrar el componente correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe de tener un parrafo con el titulo', () => {
        
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe(tittle);


    });

    test('debe de tener la imagen igual al url y alt de los props', () => {

        const img = wrapper.find('img');

        expect( img.prop('src') ).toBe(url);
        expect( img.prop('alt') ).toBe(tittle);

    });

    test('debe de tener animate__fadeIn', () => {
        
        const div = wrapper.find('div');

        const classBuscada = div.prop('className');
        
        expect( classBuscada.includes('animate__fadeIn') ).toBe( true );

    })
    
    
    
    


})

