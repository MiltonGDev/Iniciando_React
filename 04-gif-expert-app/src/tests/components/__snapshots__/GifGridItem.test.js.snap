// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para GifGridItem debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="A tittle"
    src="https://media2.giphy.com/media/yo3TC0yeHd53G/giphy.gif?cid=446fbf6ca4d0d24cygq8kkdjokh6d3w8qdz4otn5u21ealob&rid=giphy.gif&ct=g"
  />
  <p>
    A tittle
  </p>
</div>
`;
